src/foam/nanos/approval/ApprovalRequest.js

```diff
@@ -542,59 +540,54 @@
     {
       name: 'approve',
       section: 'requestDetails',
-      isAvailable: (isTrackingRequest, status) => {
-        if (
-          status === foam.nanos.approval.ApprovalStatus.REJECTED ||
-          status === foam.nanos.approval.ApprovalStatus.APPROVED ||
-          status === foam.nanos.approval.ApprovalStatus.CANCELLED
-        ) {
-          return false;
-        }
+      isAvailable: function(isTrackingRequest, status) {
+        if ( status !== this.ApprovalStatus.REQUESTED ) return false;
         return ! isTrackingRequest;
       },
       code: function(X) {
-        this.approve.bind(this);
+        var approvedApprovalRequest = this.clone();
+        approvedApprovalRequest.status = this.ApprovalStatus.APPROVED;
+
+        this.approvalRequestDAO.put(approvedApprovalRequest).then(req => {
+          this.approvalRequestDAO.cmd(this.AbstractDAO.RESET_CMD);
+          this.finished.pub();
+          this.notify(this.SUCCESS_APPROVED, '', this.LogLevel.INFO, true);
+          X.stack.back();
+        }, e => {
+          this.throwError.pub(e);
+          this.notify(e.message, '', this.LogLevel.ERROR, true);
+        });
       }
     },
     {
       name: 'approveWithMemo',
       section: 'requestDetails',
-      isAvailable: (isTrackingRequest, status) => {
-        if (
-          status === foam.nanos.approval.ApprovalStatus.REJECTED ||
-          status === foam.nanos.approval.ApprovalStatus.APPROVED ||
-          status === foam.nanos.approval.ApprovalStatus.CANCELLED
-        ) {
-          return false;
-        }
+      isAvailable: function(isTrackingRequest, status) {
+        if ( status !== this.ApprovalStatus.REQUESTED ) return false;
         return ! isTrackingRequest;
       },
       code: function(X) {
-        var objToAdd = X.objectSummaryView ? X.objectSummaryView : X.summaryView;
+        var objToAdd = X.objectSummaryView ?
+          X.objectSummaryView : X.summaryView;
         objToAdd.add(this.Popup.create({ backgroundColor: 'transparent' }).tag({
-          class: "foam.u2.MemoModal",
-          onExecute: this.approveWithMemo.bind(this, X)
+          class: 'foam.u2.MemoModal',
+          onExecute: this.approveWithMemoL.bind(this, X)
```

Comment:
approveWithMemoL ?

Reply:
Listener - I think its terribly confusing having an action and listener with the same name.

Reply:
You can't have them with the same name. FOAM might not detect the error, but things won't work correctly.
You don't need to bind() listeners to this. They're pre-bound.